_posts/2024-07-03-oidc-cient-request.adoc

```diff
@@ -0,0 +1,447 @@
+* Under *Store password* enter the password of the keystore you created
+and click on the *Browse* button to import the keystore file. Note that when we created the keystore, we specified a key
+password and a keystore password. Here, we are only using the *keystore* password. While we set them both to be the same, they do not have to be.
+* Once you select the keystore file created earlier, click *Import* and you should see a message at the top of the screen indicating that the certificate has been uploaded successfully and you will see it in the text field in the middle of the screen.
+
+. Finally, create a user called *alice* whose *First name* is *Alice* and *Last name* is *Smith*. You can find more details about creating and managing KeyCloak users https://www.keycloak.org/docs/latest/server_admin/#proc-creating-user_server_administration_guide[here].
+
+. Once you hit *Create* to create the user, navigate to the *Credentials* tab at the top of the page and create a password for Alice. Toggle the `Temporary` switch off, so you are not prompted to update the password after the first login.
+
+== Create an OpenShift Secret
+
+Since WildFly will use the keystore we created earlier, we need to add it to OpenShift. We can do this by generating an OpenShift secret using the keystore file as follows: 
+```
+    $ oc create secret generic simple-webapp-secret --from-file=/PATH/TO/Keycloak.keystore.pkcs12
+```
+
+== Add Helm Configuration
+* Obtain the URL for Keycloak.
+```
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak URL:   $KEYCLOAK_URL" &&
+echo ""
+```
+
+* Switch to the charts directory in the `elytron-oidc-client-scope` example.
+```
+    $ cd /PATH/TO/ELYTRON/EXAMPLES/elytron-oidc-request/charts
+```
+Notice there’s a helm.yaml file in this directory with the following content:
+```
+build:
+  uri: https://github.com/wildfly-security-incubator/elytron-examples.git
+  contextDir: elytron-oidc-request
+deploy:
+  replicas: 1
+  volumes:
+    - name: saml-keystore-volume
+      secret:
+        secretName: simple-webapp-saml-secret
+  env:
+    - name: OIDC_PROVIDER_URL
+      value: <KEYCLOAK_URL>                         <1>
+    - name: OIDC_CLIENT_SECRET
+      value: <CLIENT_SECRET>                        <2>
+    - name: AUTH_REQUEST_FORMAT
+      value: request
+    - name: SIGNING_KEYSTORE_PATH                   
+      value: /etc/request-object-secret-volume/Keycloak.keystore.pkcs12     
+    - name: SERVER_ARGS
+      value: "--stability=preview" 
+  volumes:
+    - name: request-object-signing-keystore-volume
+      secret:
+        secretName: simple-webapp-secret
+  volumeMounts:
+    - name: request-object-signing-keystore-volume
+      mountPath: /etc/request-object-secret-volume
+      readOnly: true
+```
+<1> Replace <KEYCLOAK_URL> with the Keycloak URL obtained in the previous command.
+<2> Replace <CLIENT_SECRET> with the client secret for `myclient`.
```

Comment:
I don't understand what to designate for the "CLIENT_SECRET".   Here was no previous direction to create a "client secrete".  There was a direction to create an "OpenShift secret".  There was not direction to save the secret text for future reference.

This is blocking me from proceeding with the example.

Reply:
Thanks @rsearls 
I have addressed the comments till here.